soroban-env-host/src/host.rs

```diff
@@ -618,6 +619,19 @@ impl Host {
         }
     }
 
+    #[cfg(any(test, feature = "testutils"))]
+    pub fn with_debug_mode<F>(&self, f: F)
```

Comment:
I've commented on the other PR (sdk), not sure why this needs to be exposed.

Reply:
Is the only difference between these functions the `pub` vs `pub(crate)`?

Reply:
yeah

Reply:
I changed things around to just expose a method that pull diagnostics.